mne_bids/write.py

```diff
@@ -976,16 +978,18 @@ def write_raw_bids(raw, bids_basename, bids_root, events_data=None,
     # "invalid" basename for an emptyroom subject.
     on_invalid_er_task = 'continue'
 
-    scans_fname = _gen_bids_basename(
-        subject=subject_id, session=session_id, suffix='scans.tsv',
-        prefix=ses_path, on_invalid_er_task=on_invalid_er_task)
+    scans_fname = _gen_bids_basename(sub=subject_id, ses=session_id,
+                                     prefix=ses_path, suffix='scans.tsv',
+                                     on_invalid_er_task=on_invalid_er_task)
     coordsystem_fname = _gen_bids_basename(
-        subject=subject_id, session=session_id, acquisition=acquisition,
-        suffix='coordsystem.json', prefix=data_path,
+        sub=subject_id, ses=session_id,
+        acq=acquisition, prefix=data_path,
+        suffix='coordsystem.json',
         on_invalid_er_task=on_invalid_er_task)
     electrodes_fname = _gen_bids_basename(
-        subject=subject_id, session=session_id, acquisition=acquisition,
-        suffix='electrodes.tsv', prefix=data_path,
+        sub=subject_id, ses=session_id,
+        acq=acquisition, prefix=data_path,
+        suffix='electrodes.tsv',
```

Comment:
all this code could benefit from the `BIDSPath` api. And also below. Can you do that? Like in my original PR

Reply:
How should wee handle the `on_valid_er_task`? I'm not too familiar w/ what's going on there. Can we just stick that kwarg into `BIDSPath`?

Reply:
no it should be handled outside `BIDSPath`

Reply:
Okay, I kept these here that use the `_gen_bids_basename` for `on_valid_er_task`, but handled some of the below code.

Reply:
you might want @hoechenberger 's opinion here.